integration/setup/setup.go

```diff
@@ -62,3 +80,248 @@ func Setup(ctx context.Context, c Config) error {
+    Name: ci-azure-operator
+    Provider:
+      Azure:
+        Cloud: AZUREPUBLICCLOUD
+        HostCluster:
+          ResourceGroup: godsmack
+          VirtualNetwork: "godsmack"
+          VirtualNetworkGateway: "godsmack-vpn-gateway"
+          CIDR: "0.0.0.0/0"
+        MSI:
+          Enabled: true
+        Location: %s
+    Registry:
+      Domain: quay.io
+    Secret:
+      AzureOperator:
+        SecretYaml: |
+          service:
+            azure:
+              clientid: "%s"
+              clientsecret: "%s"
+              subscriptionid: "%s"
+              tenantid: "%s"
+              template:
+                uri:
+                  version: %s
+`
+		defer func() {
+			fs := afero.NewOsFs()
+			err := fs.Remove(operatorTarballPath)
+			if err != nil {
+				config.Logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("deletion of %#q failed", operatorTarballPath), "stack", fmt.Sprintf("%#v", err)) // nolint: errcheck
+			}
+		}()
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("installing %#q", project.Name())) // nolint: errcheck
+
+		err = config.HelmClient.InstallReleaseFromTarball(ctx,
+			operatorTarballPath,
+			key.Namespace(),
+			helm.ReleaseName(key.ReleaseName()),
+			helm.ValueOverrides([]byte(fmt.Sprintf(values, env.AzureClientID(), env.AzureTenantID(), env.AzureLocation(), env.AzureClientID(), env.AzureClientSecret(), env.AzureSubscriptionID(), env.AzureTenantID(), env.CircleSHA()))),
+			helm.InstallWait(true))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("installed %#q", project.Name())) // nolint: errcheck
+	}
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "ensuring chart CRD exists") // nolint: errcheck
+
+		// The operator will install the CRD on boot but we create chart CRs
+		// in the tests so this ensures the CRD is present.
+		err = config.K8sClients.CRDClient().EnsureCreated(ctx, appv1alpha1.NewChartCRD(), backoff.NewMaxRetries(7, 1*time.Second))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "ensured chart CRD exists") // nolint: errcheck
+	}
+
+	{
+		encryptionSecret := &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-%s", key.TestAppReleaseName(), "encryption"),
+				Namespace: "default",
+				Labels: map[string]string{
+					"giantswarm.io/cluster":   env.ClusterID(),
+					"giantswarm.io/randomkey": "encryption",
+				},
+			},
+			Data: map[string][]byte{
+				"encryption": []byte("B+QdiVynV8Z6bgo1TpAD6Qj1DvRTAx2j/j9EoNlOP38="),
+			},
+			Type: "Opaque",
+		}
+
+		_, err := config.K8sClients.K8sClient().CoreV1().Secrets("default").Create(encryptionSecret)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	{
+		azureConfig := &providerv1alpha1.AzureConfig{
```

Comment:
This is one of the highlights: we create the `AzureConfig` using client-go instead of using the e2e test helm chart in the apiextensions repository.